src/media/headset.ts

```diff
@@ -0,0 +1,70 @@
+import { Observable } from 'rxjs';
+import GenesysCloudWebrtcSdk from '../client';
+import HeadsetService, { ConsumedHeadsetEvents, VendorImplementation} from 'softphone-vendor-headsets';
+
+export class SdkHeadset {
+    private sdk: GenesysCloudWebrtcSdk;
+    private headsetLibrary: HeadsetService;
+    headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+    constructor(sdk: GenesysCloudWebrtcSdk) {
+        this.sdk = sdk;
+        this.headsetLibrary = HeadsetService.getInstance({ logger: sdk.logger });
+        this.headsetEvents$ = this.headsetLibrary.headsetEvents$;
+    }
+
+    updateAudioInputDevice (newMicId: string): void {
+        const completeDeviceInfo = this.sdk.media.findCachedDeviceByIdAndKind(newMicId, 'audioinput');
+        this.headsetLibrary.activeMicChange(completeDeviceInfo.label.toLowerCase());
+    }
+
+    getCurrentSelectedImplementation (): VendorImplementation {
+        return this.headsetLibrary.selectedImplementation;
+    }
+
+    showRetry (): boolean {
+        const selectedImplementation = this.getCurrentSelectedImplementation();
+        if (selectedImplementation?.disableRetry) {
+            return false;
+        }
+
+        return !!selectedImplementation
+            && !selectedImplementation.isConnected
+            && !selectedImplementation.isConnecting;
+    }
+
+    retryConnection (micLabel: string): void {
+        this.headsetLibrary.retryConnection(micLabel);
+    }
+
+    setRinging (callInfo: { conversationId: string, contactName?: string }, hasOtherActiveCalls): Promise<void> {
+        return this.headsetLibrary.incomingCall(callInfo, hasOtherActiveCalls);
+    }
+
+    outgoingCall (callInfo: { conversationId: string, contactName: string }): Promise<void> {
+        return this.headsetLibrary.outgoingCall(callInfo);
+    }
+
+    endCurrentCall (conversationId: string): Promise<void> {
+        if (conversationId) {
+            return this.headsetLibrary.endCall(conversationId);
+        }
+    }
```

Comment:
This function is going to be tricky because you are only return a promise _if_ a `conversationId` is passed in. This could cause bugs if consumers call `endCurrentCall().then()` which will throw: `Uncaught TypeError: Cannot read properties of undefined (reading 'then')`. 

Two quick solutions: 
1. make it async: `endCurrentCall (conversationId: string): Promise<void>`
2. add a `return Promise.resolve()` at the end.

Reply:
@maxwellmooney13  You actually don't need to promise.resolve if the function is marked async, it will always return a promise.